vignettes/runif-via-pmap.Rmd

```diff
@@ -0,0 +1,213 @@
+  pmap(runif)
+```
+
+One can also create a wrapper around `runif()` that does the correspondance between column names and argument names on the fly.
+
+```{r}
+# 1st option of the wrapper
+my_runif <- function(...) {
+  l <- list(...)
+  runif(n = l$alpha, min = l$beta, max = l$gamma)
+}
+
+set.seed(123)
+pmap(foofy, my_runif)
+```
+
+One can also pass the arguments directly as vectors.
+
+```{r}
+my_runif <- function(alpha, beta, gamma, ...) {
+  runif(n = alpha, min = beta, max = gamma)
+}
+set.seed(123)
+pmap(foofy, my_runif)
+```
+ 
+A third option that comes in handy is to refer to inputs by their position `..i`. Yet, one should be careful when inputting the column numbers
+
+```{r}
+set.seed(123)
+pmap(foofy, ~ runif(n = ..1, min = ..2, max = ..3))
+```
+
+### Extra variables in the data frame
+
+One situation that comes up so often is having a dataframe that includes columns that shouldn't be passed to the function `.f()`. Let us create a tibble to illustrate this.
+
+```{r}
+df_oops <- tibble(
+  n = 1:3,
+  min = c(0, 10, 100),
+  max = c(1, 100, 1000),
+  oops = c("please", "ignore", "me")
+)
+df_oops
+```
+
+Passing the dataframe as it is is not going to work !
+
+```{r error=TRUE}
+set.seed(123)
+pmap(df_oops, runif)
+```
+
+Selecting the specific columns that should be passed to `.f()` or ommitting the columns that shouldn't be passed is a good way of doing. This can be done using `dplyr::select()`.
+
+```{r}
+set.seed(123)
+df_oops %>%
+  select(n, min, max) %>% ## if it's easier to say what to keep
+  pmap(runif)
+
+# OR
+
+set.seed(123)
+df_oops %>%
+  select(-oops) %>%       ## if it's easier to say what to omit
+  pmap(runif)
+```
+
+A custom wrapper with the dot syntax `...` can also get the job done.
+
+```{r}
+my_runif <- function(n, min, max, ...) runif(n, min, max)
+
+set.seed(123)
+pmap(df_oops, my_runif)
+```
+
+### Add the generated data to the data frame as a list-column
+
+The generated data can be embedded inside the dataframe using `mutate()`.
+
+```{r}
+set.seed(123)
+(df_aug <- df %>%
+    mutate(data = pmap(., runif)))
+```
+
+This can still be done if the name of the columns aren't the same as the name of the function arguments. `list()` should be used instead of the placeholder `.`. Variables can then be renamed inside the list.
+
+```{r}
+foofy <- tibble(
+  alpha = 1:3,            ## was: n
+  beta = c(0, 10, 100),   ## was: min
+  gamma = c(1, 100, 1000) ## was: max
+)
+
+set.seed(123)
+foofy %>%
+  mutate(data = pmap(list(n = alpha, min = beta, max = gamma), runif))
+```
+
+`list()`can also be used to select only the variables that are needed or to exclude only some of the variables.
+
+```{r}
+df_oops <- tibble(
+  n = 1:3,
+  min = c(0, 10, 100),
+  max = c(1, 100, 1000),
+  oops = c("please", "ignore", "me")
+)
+
+set.seed(123)
+df_oops %>%
+  mutate(data = pmap(list(n, min, max), runif))
+
+# Excluding "oops" column
+df_oops %>%
+  mutate(data = pmap(select(., -oops), runif))
+```
+
+### Review
```

Comment:
I think this would make sense at the beginning of the doc to give an overview of what will be covered.

Reply:
Made some fixes